validators/src/unit-test/java/uk/gov/moj/sdt/validators/BulkSubmissionValidatorTest.java

```diff
@@ -465,46 +436,38 @@ public void testRequestCountdoesNotMatch() {
             errorMessage.setErrorCode(IErrorMessage.ErrorCode.REQ_COUNT_MISMATCH.name());
             errorMessage
                     .setErrorText("Unexpected Total Number of Requests identified. {0} requested identified, {1} requests expected in Bulk Request {2}.");
-            errorMessagesCache = EasyMock.createMock(ICacheable.class);
-            expect(
-                    errorMessagesCache.getValue(IErrorMessage.class,
-                            IErrorMessage.ErrorCode.REQ_COUNT_MISMATCH.name())).andReturn(errorMessage);
-            replay(errorMessagesCache);
+            when(errorMessagesCache.getValue(IErrorMessage.class,
+                            IErrorMessage.ErrorCode.REQ_COUNT_MISMATCH.name())).thenReturn(errorMessage);
             validator.setErrorMessagesCache(errorMessagesCache);
 
             bulkSubmission.accept(validator, null);
 
-            Assert.fail("Test failed to throw RequestCountMismatchException ");
+            Assertions.fail("Test failed to throw RequestCountMismatchException ");
         } catch (final RequestCountMismatchException e) {
-            LOGGER.debug(e.getMessage().toString());
-
-            EasyMock.verify(mockIBulkCustomerDao);
-            EasyMock.verify(mockIBulkSubmissionDao);
-            Assert.assertTrue("Error code incorrect",
-                    e.getErrorCode().equals(IErrorMessage.ErrorCode.REQ_COUNT_MISMATCH.name()));
-            Assert.assertTrue(
-                    "Substitution value incorrect",
-                    e.getErrorDescription().equals(
+            LOGGER.debug(e.getMessage());
+
+            verify(mockIBulkCustomerDao).getBulkCustomerBySdtId(12345L);
+            verify(mockIBulkSubmissionDao).getBulkSubmission(bulkCustomer, bulkSubmission.getCustomerReference(),
+                    DATA_RETENTION_PERIOD);
+            Assertions.assertTrue(e.getErrorCode().equals(IErrorMessage.ErrorCode.REQ_COUNT_MISMATCH.name()),
+                    ERROR_CODE_INCORRECT);
+            Assertions.assertTrue(e.getErrorDescription().equals(
                             "Unexpected Total Number of Requests identified. 1 requested identified, " + mismatchTotal +
                                     " requests expected in Bulk Request " + bulkSubmission.getCustomerReference() +
-                                    "."));
+                                    "."),
+                    SUBSTITUTION_VALUE_INCORRECT);
         }
-
     }
 
     /**
      * This test the scenario where all individual requests of a bulk submission are rejected.
      */
     @Test
-    public void testIndividualRequestAllRejected() {
+    void testIndividualRequestAllRejected() {
         // set up the data we are going to use for this customer
         // set up bulk customer with the application it can use
         bulkCustomer = createCustomer(createBulkCustomerApplications("MCOL"));
 
```

Comment:
Set up of mocking behaviour of mockIBulkCustomerDao has been removed from this section.  Is it needed?

Reply:
Removed deliberately: Unnecessary stubbings detected.
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at uk.gov.moj.sdt.validators.BulkSubmissionValidatorTest.testIndividualRequestAllRejected